platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/preconsensus/PcesReplayer.java

```diff
@@ -67,19 +71,21 @@ public class PcesReplayer {
      *                                 overwhelmed
      */
     public PcesReplayer(
-            final @NonNull Time time,
+            final @NonNull PlatformContext context,
             final @NonNull StandardOutputWire<PlatformEvent> eventOutputWire,
             final @NonNull Runnable flushIntake,
             final @NonNull Runnable flushTransactionHandling,
             final @NonNull Supplier<ReservedSignedState> latestImmutableState,
             final @NonNull Supplier<Boolean> isSystemHealthy) {
 
-        this.time = Objects.requireNonNull(time);
+        this.time = context.getTime();
         this.eventOutputWire = Objects.requireNonNull(eventOutputWire);
         this.flushIntake = Objects.requireNonNull(flushIntake);
         this.flushTransactionHandling = Objects.requireNonNull(flushTransactionHandling);
         this.latestImmutableState = Objects.requireNonNull(latestImmutableState);
         this.isSystemHealthy = Objects.requireNonNull(isSystemHealthy);
+
+        this.config = context.getConfiguration().getConfigData(PcesConfig.class);
```

Comment:
This approach hides the coupling between the two pieces.
It's been used in many places and I know that this pattern simplifies changes to the code, but I found it challenging in other scenarios, e.g., (when working with the FileSystemManager) to test without knowing objects' direct dependencies and relationships.

Reply:
Is there a different pattern you're advocating for?

I'm not particularly attached to this way of doing things, but we should probably reach internal team agreement on a preferred strategy if we want to change it.

Reply:
IMO, the higher the layer we can push the usage of the configs object, the better. And after a certain point, the config framework should no longer be used, and all values should have been retrieved. 

One problem I faced was that when testing something else and needing these configuration objects, It was not enough to mock the constructor parameters; I needed to go to the implementation and see how the config framework was being used to mock the config framework, too. That defeats the black box testing strategy. 

 This would also help to provide fast failures, as any necessary value that is not retrieved after a point would produce a failure instead of waiting for it to be used to fail. Stretching the argument a little more, the more the configuration framework is used inside the inner layers, the more difficult it would be to replace it in the future if needed.
 
 But I am stating this from a personal point of view, given that I don't know if there is a place that can become the boundary where all the configs are retrieved.

Reply:
I agree, it is often painful to conclude which config values affect testing.

Perhaps you could start a conversation with the rest of the platform team, to decide if we need to shift our general strategy for using config values?